srplasticity/srp_extrafuncs.py

```diff
@@ -304,11 +324,29 @@ def plot_mse_fig(axis, mses):
         size=11,
         weight="bold",
         usetex=False,
-        family="calibri",
     )
 
 
 def gen_kernel(mu_amps, mu_taus, mu_baseline=None, dt=1):
+    """
```

Comment:
What exactly is this function used for? We have now formalized easySRP as a class so if this function is used to present the kernel from an existing fitted model the user can simple call "model_name.mu_kernel" to access the kernel attribute from the fitted model and then use "srp.convolve_spiketrain_with_kernel". Unless something else is going on I think this function is now depreciated.

Reply:
I was using it to plot the kernel itself in plot_kernel (i.e., get the x_vals and y_vals). I could probably make it shorter using model.mu_kernel, but I would still need to arrange it so that it can be used directly by plot_kernel (how just add the code to plot_kernel). It could probably be replaced by something like this, where I use x_vals and y_vals:
![Capture d’écran 2024-11-30 122516](https://github.com/user-attachments/assets/4ec34cbe-3b2e-498c-b2ff-d6ae94d58cab)

Reply:
Most of this behavior is already handled by tools.get_stim_vec  and srp.convolve_spiktrain_with_kernel through run_spiketrain/run_ISIvec methods of the SRP model classes. If you still need more than two lines (which could just be a block in the protocol+notebook) then I would package them inside plot_kernel or in a function named something like arrange_kernel. I think it will be confusing for users if we duplicate the existing kernel generation infrastructure with a gen_kernel function.

Reply:
For now, I'll change the name to _arrange_kernel (since it's only supposed to be used by plot_kernel) and make a comment in the docstring about its purpose.